From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kanaka Raju Nayana <kanaka.raju.nayana@intel.com>
Date: Wed, 29 Oct 2025 09:03:30 +0530
Subject: Add debug print and BACKPORT_MOD_VERSIONS for oot modules

Add debug prints in vsec and pmt files to identify
the backport driver.

Add BACKPORT_MOD_VERSIONS to display information
about oot module version.

V1: update with backports tag [Raju]

Signed-off-by: Kanaka Raju Nayana <kanaka.raju.nayana@intel.com>
Signed-off-by: Akanksha Hotta <akanksha.hotta@intel.com>
---
 drivers/platform/x86/intel/pmt/class.c     | 2 ++
 drivers/platform/x86/intel/pmt/crashlog.c  | 2 ++
 drivers/platform/x86/intel/pmt/telemetry.c | 2 ++
 drivers/platform/x86/intel/vsec.c          | 2 ++
 4 files changed, 8 insertions(+)

diff --git a/drivers/platform/x86/intel/pmt/class.c b/drivers/platform/x86/intel/pmt/class.c
index 93edaa7..1f89907 100644
--- a/drivers/platform/x86/intel/pmt/class.c
+++ b/drivers/platform/x86/intel/pmt/class.c
@@ -433,6 +433,7 @@ EXPORT_SYMBOL_NS_GPL(intel_pmt_dev_destroy, "INTEL_PMT");
 
 static int __init pmt_class_init(void)
 {
+	printk(KERN_INFO "VSEC CLASS Backport init, Module is backported from " CPTCFG_BACKPORTS_RELEASE_TAG "\n");
 	return class_register(&intel_pmt_class);
 }
 
@@ -444,6 +445,7 @@ static void __exit pmt_class_exit(void)
 module_init(pmt_class_init);
 module_exit(pmt_class_exit);
 
+BACKPORT_MOD_VERSIONS
 MODULE_AUTHOR("Alexander Duyck <alexander.h.duyck@linux.intel.com>");
 MODULE_DESCRIPTION("Intel PMT Class driver");
 MODULE_LICENSE("GPL v2");
diff --git a/drivers/platform/x86/intel/pmt/crashlog.c b/drivers/platform/x86/intel/pmt/crashlog.c
index 7b657ff..8ee1194 100644
--- a/drivers/platform/x86/intel/pmt/crashlog.c
+++ b/drivers/platform/x86/intel/pmt/crashlog.c
@@ -685,6 +685,7 @@ static struct auxiliary_driver pmt_crashlog_aux_driver = {
 
 static int __init pmt_crashlog_init(void)
 {
+	printk(KERN_INFO "VSEC CRASHLOG Backport init, Module is backported from " CPTCFG_BACKPORTS_RELEASE_TAG "\n");
 	return auxiliary_driver_register(&pmt_crashlog_aux_driver);
 }
 
@@ -697,6 +698,7 @@ static void __exit pmt_crashlog_exit(void)
 module_init(pmt_crashlog_init);
 module_exit(pmt_crashlog_exit);
 
+BACKPORT_MOD_VERSIONS
 MODULE_AUTHOR("Alexander Duyck <alexander.h.duyck@linux.intel.com>");
 MODULE_DESCRIPTION("Intel PMT Crashlog driver");
 MODULE_LICENSE("GPL v2");
diff --git a/drivers/platform/x86/intel/pmt/telemetry.c b/drivers/platform/x86/intel/pmt/telemetry.c
index 6999121..cff89fc 100644
--- a/drivers/platform/x86/intel/pmt/telemetry.c
+++ b/drivers/platform/x86/intel/pmt/telemetry.c
@@ -428,6 +428,7 @@ static struct auxiliary_driver pmt_telem_aux_driver = {
 
 static int __init pmt_telem_init(void)
 {
+	printk(KERN_INFO "VSEC TELEMETRY Backport init, Module is backported from " CPTCFG_BACKPORTS_RELEASE_TAG "\n");
 	return auxiliary_driver_register(&pmt_telem_aux_driver);
 }
 module_init(pmt_telem_init);
@@ -439,6 +440,7 @@ static void __exit pmt_telem_exit(void)
 }
 module_exit(pmt_telem_exit);
 
+BACKPORT_MOD_VERSIONS
 MODULE_AUTHOR("David E. Box <david.e.box@linux.intel.com>");
 MODULE_DESCRIPTION("Intel PMT Telemetry driver");
 MODULE_LICENSE("GPL v2");
diff --git a/drivers/platform/x86/intel/vsec.c b/drivers/platform/x86/intel/vsec.c
index 439c0c8..937214f 100644
--- a/drivers/platform/x86/intel/vsec.c
+++ b/drivers/platform/x86/intel/vsec.c
@@ -614,6 +614,7 @@ static bool intel_vsec_walk_vsec(struct pci_dev *pdev,
 int intel_vsec_register(struct device *dev,
 			const struct intel_vsec_platform_info *info)
 {
+	printk(KERN_INFO "VSEC Backport init, Module is backported from " CONFIG_BACKPORTS_RELEASE_TAG "\n");
 	if (!dev || !info || !info->headers)
 		return -EINVAL;
 
@@ -902,6 +903,7 @@ static struct pci_driver intel_vsec_pci_driver = {
 };
 module_pci_driver(intel_vsec_pci_driver);
 
+BACKPORT_MOD_VERSIONS
 MODULE_AUTHOR("David E. Box <david.e.box@linux.intel.com>");
 MODULE_DESCRIPTION("Intel Extended Capabilities auxiliary bus driver");
 MODULE_LICENSE("GPL v2");
-- 
2.43.0
